benchmark/README.md

```diff
@@ -1,147 +1,283 @@
+arrays/zero-int.js n=25 type=Buffer: 90.49906662339653
 ...
 ```
 
-The last number is the rate of operations. Higher is better.
+It is possible to execute more groups by adding extra process arguments.
+```
+$ node benchmark/run.js arrays buffers
+```
 
-### Run an individual test
+### Comparing node versions
 
-For example, buffer-slice.js:
+To compare the effect of a new node version use the `compare.js` tool. This
+will run each benchmark multiple times, making it possible to calculate
+statistics on the performance measures.
+
+As an example on how to check for a possible performance improvement, the
+[#5134](https://github.com/nodejs/node/pull/5134) pull request will be used as
+an example. This pull request _claims_ to improve the performance of the
+`string_decoder` module.
+
+First build two versions of node, one from the master branch (here called
+`./node-master`) and another with the pull request applied (here called
+`./node-pr-5135`).
+
+The `compare.js` tool will then produce a csv file with the benchmark results.
 
-```bash
-node benchmark/buffers/buffer-read.js
 ```
-The output:
+$ node benchmark/compare.js string_decoder --old ./node-master --new ./node-pr-5134 > compare-pr-5134.csv
+```
+
+For analysing the benchmark results use the `compare.R` tool.
+
 ```
-buffers/buffer-read.js noAssert=false buffer=fast type=UInt8 millions=1: 246.79
-buffers/buffer-read.js noAssert=false buffer=fast type=UInt16LE millions=1: 240.11
-buffers/buffer-read.js noAssert=false buffer=fast type=UInt16BE millions=1: 245.91
+$ cat compare-pr-5134.csv | Rscript benchmark/compare.R
+
+                                                                                      improvement significant      p.value
+string_decoder/string-decoder.js n=250000 chunk=1024 inlen=1024 encoding=ascii           12.46 %         *** 1.165345e-04
+string_decoder/string-decoder.js n=250000 chunk=1024 inlen=1024 encoding=base64-ascii    24.70 %         *** 1.820615e-15
+string_decoder/string-decoder.js n=250000 chunk=1024 inlen=1024 encoding=base64-utf8     23.60 %         *** 2.105625e-12
+string_decoder/string-decoder.js n=250000 chunk=1024 inlen=1024 encoding=utf8            14.04 %         *** 1.291105e-07
+string_decoder/string-decoder.js n=250000 chunk=1024 inlen=128  encoding=ascii            6.70 %           * 2.928003e-02
 ...
 ```
 
-### Run tests with options
+In the output _improvement_ is the relative improvement of the new version,
+hopefully this is positive. _significant_ tells if there are enough
+statistical evidence to validate the _improvement_. If there are enough evidence
+then there will be at least one star (`*`), more stars is just better. **However
+if there are no stars, then you shouldn't make any conclusions based on the
+_improvement_.** Sometimes this is fine, for example if you are expecting there
+to be no improvements, then there shouldn't be any stars.
+
+_For the statistically minded the R scripts makes an [independent/unpaired
```

Comment:
So, I’m not a native speaker, but I would say there should be a comma after `,`; otherwise, the sentence sounds like the R script does what it does *for specific people*. :)

The `scripts` -> `script` change suggestion is only to make the noun match the singular verb, and `makes` -> `performs` is just a suggestion based on how I think most people use the language around statistical tests.